libjulia/backends/evm/EVMCodeTransform.cpp

```diff
@@ -51,10 +99,19 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 		while (variablesLeft--)
 			m_assembly.appendConstant(u256(0));
 	}
-	for (auto const& variable: _varDecl.variables)
+	for (auto const& variable: _varDecl.variables | boost::adaptors::reversed)
 	{
 		auto& var = boost::get<Scope::Variable>(m_scope->identifiers.at(variable.name));
-		m_context->variableStackHeights[&var] = height++;
+		if (!m_context->unusedStackSlots.empty()) {
```

Comment:
If there were no unused stack slots at the very beginning, this mode is rather wasteful - because it reverses the whole value list. I think the optimal way would be to start going through the variables in reverse until there are no unused stack slots anymore, at which point we should go through the list in normal order - but that might be a little complicated to implement.